lib/ansible/modules/cloud/amazon/ec2_vpc_igw.py

```diff
@@ -163,7 +163,7 @@ def ensure_tags(self, igw_id, tags, add_only):
                     # update tags
                     final_tags.update(to_update)
                 else:
-                    AWSRetry.exponential_backoff()(self._connection.create_tags)(
+                    AWSRetry.exponential_backoff(catch_extra_error_codes=['InvalidInternetGatewayID.NotFound'])(self._connection.create_tags)(
```

Comment:
Here too - a waiter after resource creation would be nicer in this module though there's no built-in one. Will add in a bit.